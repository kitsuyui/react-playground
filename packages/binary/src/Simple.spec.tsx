import { describe, it, expect } from 'vitest'

import { render } from '@testing-library/react'

import {
  Simple8BitString,
  SimpleBit,
} from './Simple'

describe('Simple8BitString component', () => {
  it('should render properly with 8 bits', () => {
    const value = 0b10101010
    const { asFragment } = render(<Simple8BitString value={value} />)
    expect(asFragment()).toMatchSnapshot()
    expect(asFragment().textContent).toBe('10101010')
  })

  it('should not render overflow bits', () => {
    const value = 0b110101010
    const { asFragment } = render(<Simple8BitString value={value} />)
    expect(asFragment()).toMatchSnapshot()
    expect(asFragment().textContent).toBe('10101010')
  })
})

describe('SimpleBit component', () => {
  it('should render "1" for stand=true and overflow=false', () => {
    const { asFragment } = render(<SimpleBit stand={true} overflow={false} littleEndianBitOrder={0} bigEndianBitOrder={0} />)
    expect(asFragment()).toMatchSnapshot()
    expect(asFragment().textContent).toBe('1')
  })

  it('should not render overflow bits', () => {
    const { asFragment } = render(<SimpleBit stand={true} overflow={true} littleEndianBitOrder={0} bigEndianBitOrder={0} />)
    expect(asFragment()).toMatchSnapshot()
    expect(asFragment().textContent).toBe('')
  })

  it('should treat bit does\'t overflow by default', () => {
    const { asFragment } = render(<SimpleBit stand={true} littleEndianBitOrder={0} bigEndianBitOrder={0} />)
    expect(asFragment()).toMatchSnapshot()
    expect(asFragment().textContent).toBe('1')
  })
})
