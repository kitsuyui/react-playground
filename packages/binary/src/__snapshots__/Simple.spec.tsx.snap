// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Simple8BitString component > should not render overflow bits 1`] = `
<DocumentFragment>
  10101010
</DocumentFragment>
`;

exports[`Simple8BitString component > should render properly with 8 bits 1`] = `
<DocumentFragment>
  10101010
</DocumentFragment>
`;

exports[`SimpleBit component > should not render overflow bits 1`] = `<DocumentFragment />`;

exports[`SimpleBit component > should render "1" for stand=true and overflow=false 1`] = `
<DocumentFragment>
  1
</DocumentFragment>
`;

exports[`SimpleBit component > should treat bit does't overflow by default 1`] = `
<DocumentFragment>
  1
</DocumentFragment>
`;
